mujoco_warp/_src/collision_driver_test.py

```diff
@@ -288,6 +390,58 @@ class CollisionTest(parameterized.TestCase):
   #      </mujoco>
   #    """,
 
+  @parameterized.parameters(_SDF_SDF.keys())
+  def test_sdf_collision(self, fixture):
+      
+      """Tests collisions with different geometries."""
+
+      @wp.func
+      def user_sdf(p: wp.vec3, attr: wp.vec3,  sdf_type: int) -> float:
+          result = 0.0
+          if sdf_type == int(SDFType.NUT.value):
+              result = nut(p, attr)
+          elif sdf_type == int(SDFType.BOLT.value):
+              result = bolt(p, attr)
+          return result
+
+      @wp.func
+      def user_sdf_grad(p: wp.vec3, attr: wp.vec3, sdf_type: int) -> wp.vec3:
+            if sdf_type == int(SDFType.NUT.value):
+              return nut_sdf_grad(p, attr)
+            elif sdf_type == int(SDFType.BOLT.value):
+              return bolt_sdf_grad(p, attr)
+            return wp.vec3()
+
+
+      collision_sdf.user_sdf = user_sdf
```

Comment:
It there a way to make this more general, e.g. passing a dictionary of functions in order to make the model more composable? I guess not because it would imply a callback. @nimrod-gileadi for visibility

Reply:
Since - for a given run - the SDF attributes are not allowed to change dynamically, what if we ask the user to declare them as wp.static? From https://nvidia.github.io/warp/codegen.html#example-function-pointers it looks like we could then store in `Model` the index corresponding to each geom of the functions in a user-provided dict (or list) and call the appropriate one from the `sdf_eval` kernel by just knowing which geom we are dealing with.

Reply:
What about https://nvidia.github.io/warp/codegen.html#dynamic-kernel-creation